modules/fusioncatcher/fusioncatcher/main.nf

```diff
@@ -0,0 +1,64 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process FUSIONCATCHER {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::fusioncatcher=1.33" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fusioncatcher:1.33--hdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/fusioncatcher:1.33--hdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path resource
+
+    output:
+    tuple val(meta), path("*.fusioncatcher-fusion-genes.txt")   , optional:true, emit: fusions
+    tuple val(meta), path("*.fusioncatcher_summary.txt")        , optional:true, emit: summary
+    tuple val(meta), path("*.fusioncatcher.log")                               , emit: log
+    path "*.version.txt"                                                       , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def fastq    = reads.size() == 2 ? "${reads[0]},${reads[1]}" : "${reads[0]}"
+    def valid    = (resource.exists() && file("${resource}/organism.txt").exists()) ? true : false
+
+    if (valid) {
+
+        """
+        fusioncatcher \\
+            -d $resource \\
+            -i $fastq \\
+            -p $task.cpus \\
+            -o . \\
+            --aligners=star
+            --skip-blat \\
```

Comment:
Not entirely sure how we handled this elsewhere, but I think this should be passed by the user via `$options.args`.

Reply:
yes, it is better to pass via `$options.args`